homeassistant/components/ozw/cover.py

```diff
@@ -0,0 +1,72 @@
+"""Support for Z-Wave cover devices."""
+import logging
+
+from openzwavemqtt.const import CommandClass
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    DOMAIN as COVER_DOMAIN,
+    SUPPORT_CLOSE,
+    SUPPORT_OPEN,
+    SUPPORT_SET_POSITION,
+    CoverEntity,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import DATA_UNSUBSCRIBE, DOMAIN
+from .entity import ZWaveDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SUPPORTED_FEATURES_POSITION = SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_SET_POSITION
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Z-Wave Cover from Config Entry."""
+
+    @callback
+    def async_add_cover(values):
+        """Add Z-Wave Cover."""
+
+        if values.primary.command_class != CommandClass.SWITCH_MULTILEVEL:
+            _LOGGER.warning("Cover not implemented for values %s", values.primary)
+            return
+
+        async_add_entities([ZWaveCoverEntity(values)])
+
+    hass.data[DOMAIN][config_entry.entry_id][DATA_UNSUBSCRIBE].append(
+        async_dispatcher_connect(hass, f"{DOMAIN}_new_{COVER_DOMAIN}", async_add_cover)
+    )
+
+
+class ZWaveCoverEntity(ZWaveDeviceEntity, CoverEntity):
+    """Representation of a Z-Wave Cover device."""
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORTED_FEATURES_POSITION
+
+    @property
+    def is_closed(self):
+        """Return true if cover is closed."""
+        return self.values.primary.value < 5
+
+    @property
+    def current_cover_position(self):
+        """Return the current position of cover where 0 means closed and 100 is fully open."""
+        return self.values.primary.value
+
+    async def async_set_cover_position(self, **kwargs):
+        """Move the cover to a specific position."""
+        self.values.primary.send_value(kwargs[ATTR_POSITION])
```

Comment:
Don't we need to convert between the home assistant range which is 0-100 and the ozw range which is 0-99?

Reply:
Yep, makes sense, any idea how to convert it? Just sth like `min(kwargs[ATTR_POSITION], 99)`?

Reply:
We can add a helper function like this:
https://github.com/home-assistant/core/blob/7b4df98875755e8250bfbf522a29e0df0ac65329/homeassistant/components/ozw/light.py#L35-L42